test/e2e/azure_json.go

```diff
@@ -0,0 +1,113 @@
+	guidRegex            = `^[{]?[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}[}]?$`
+)
+
+// AzureJSONSecretSpecInput is the input for AzureJSONSecretSpec.
+type AzureJSONSecretSpecInput struct {
+	BootstrapClusterProxy framework.ClusterProxy
+	Namespace             *corev1.Namespace
+	ClusterName           string
+}
+
+// AzureJSONSecretSpec implements a test that verifies the contents of the /etc/kubernetes/azure.json file.
+func AzureJSONSecretSpec(ctx context.Context, inputGetter func() AzureJSONSecretSpecInput) {
+	var (
+		specName = "azure-json-secret"
+		input    AzureJSONSecretSpecInput
+	)
+
+	input = inputGetter()
+	Expect(input.Namespace).NotTo(BeNil(), "Invalid argument. input.Namespace can't be nil when calling %s spec", specName)
+	Expect(input.ClusterName).NotTo(BeEmpty(), "Invalid argument. input.ClusterName can't be empty when calling %s spec", specName)
+
+	By("checking the contents of azure.json via SSH session")
+	mgmtClient := input.BootstrapClusterProxy.GetClient()
+	clusterName, name := input.ClusterName, input.Namespace.Name
+	// Get the control plane's Machines.
+	clusterProxy := input.BootstrapClusterProxy.GetWorkloadCluster(ctx, name, clusterName)
+	Expect(clusterProxy).NotTo(BeNil(), "Failed to get cluster %s", clusterName)
+	machines := framework.GetControlPlaneMachinesByCluster(ctx, framework.GetControlPlaneMachinesByClusterInput{
+		Lister:      mgmtClient,
+		ClusterName: clusterName,
+		Namespace:   name,
+	})
+	Expect(machines).NotTo(BeEmpty())
+
+	// Get the cluster object.
+	cluster := framework.GetClusterByName(ctx, framework.GetClusterByNameInput{
+		Getter:    mgmtClient,
+		Name:      clusterName,
+		Namespace: name,
+	})
+	Expect(cluster).NotTo(BeNil())
+
+	controlPlaneEndpoint := cluster.Spec.ControlPlaneEndpoint.Host
+	// TODO: enumerate through all the machines.
+	hostname := machines[0].Spec.InfrastructureRef.Name
+	command, args := "cat", []string{"/etc/kubernetes/azure.json"}
+	f := new(bytes.Buffer)
+	Expect(retryWithTimeout(getAzureJSONInterval, getAzureJSONTimeout, func() error {
+		return execOnHost(controlPlaneEndpoint, hostname, sshPort, f, command, args...)
+	})).To(Succeed(), "Failed to execute command %s on host %s via SSH session", command, hostname)
+	config := controllers.CloudProviderConfig{}
+	Expect(json.Unmarshal(f.Bytes(), &config)).To(Succeed(), "Failed to unmarshal azure.json")
+
+	// Get the Azure cluster object.
+	azureCluster := GetAzureClusterByName(ctx, GetAzureClusterByNameInput{
+		Getter:    mgmtClient,
+		Name:      clusterName,
+		Namespace: name,
+	})
+	Expect(azureCluster).NotTo(BeNil())
+
+	// Validate required fields.
+	Expect(config.Cloud).To(Equal("AzurePublicCloud"), "Invalid cloud name")
+	Expect(config.TenantID).To(MatchRegexp(guidRegex), "Invalid tenant ID")
+	Expect(config.SubscriptionID).To(MatchRegexp(guidRegex), "Invalid subscription ID")
+	Expect(config.ResourceGroup).To(Equal(azureCluster.Spec.ResourceGroup), "Invalid resource group")
```

Comment:
you should already have access to the resource group name and location in the e2e config without having to get them from the AzureCluster object